core/range/shared/cover_and_include.rb

```diff
@@ -57,7 +57,6 @@
   it "returns true if argument is less than the last value of the range and greater than the first value" do
     (20..30).send(@method, 28).should be_true
     ('e'..'h').send(@method, 'g').should be_true
-    ("\u{999}".."\u{9999}").send @method, "\u{9995}"
```

Comment:
Nice catch!

There is a missing `should` so no check actually performed.

But titles of the proposed new cases for `include?`/`cover?` don't describe their behaviour precisely.

The `#cover?` method returns true because it compares begin/end and value with `<=>` (or `<`, `>`) methods and indeed `"\u{999}"` < `"\u{9995}"` < `"\u{9999}"`.

But `#include?` for non-ascii String begin/end uses the `#succ` method that seems alphabetic-aware and iterates from begin to end with `#succ` looking for a value. But `"\u{9999}"` and even `"\u{9995}"` isn't reachable from `"\u{999}"` - the largest reachable symbol is `"\u{9b6}"` (I suppose because `"\u{999}"` and `"\u{9995}"` belong to different alphabets). I assume `#succ` and consequently `#include?` should work well with characters of the same alphabet only.

So I would suggest to change titles of the new cases to describe the methods behaviour better